azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java

```diff
@@ -1092,11 +1092,12 @@ public void testKillBeforeStart() throws Exception {
       }
     });
 
-    this.runner.run();
+    FlowRunnerTestUtil.startThread(this.runner).join();
```

Comment:
Nice fix. The earlier this.runner.run() runs as part of the main thread itself and now it's running as a separate thread. For my education, how did we find this specific place causing problem? :)

Reply:
I added this test recently and that was the only thing different from the rest of the tests.